certora/specs/ERC20Votes.spec

```diff
@@ -262,34 +272,63 @@ rule delegate_no_frontrunning(method f) {
     assert other_votes_ == _other_votes, "delegate not contained";
 }
 
+
+
 // passes
-rule onMint() {
+rule mint_increases_totalSupply() {
+
     env e;
-    uint256 amount;
-    address account;
+    uint256 amount; address account;
 
     uint256 fromBlock = e.block.number;
-    uint224 totalVotesBefore = totalVotes();
-    uint256 totalSupplyBefore = totalSupply();
+    uint256 totalSupply_ = totalSupply();
+
+    mint(e, account, amount);
 
-    _mint(e, account, amount);
+    uint256 _totalSupply = totalSupply();
+    require _totalSupply < _maxSupply();
 
-    assert totalVotes() == totalVotesBefore, "totalVotes decreased";
-    assert getPastTotalSupply(e, fromBlock) == totalSupplyBefore , "previous total supply not saved properly";
+    assert _totalSupply == totalSupply_ + amount, "totalSupply not increased properly";
+    assert getPastTotalSupply(e, fromBlock) == totalSupply_ , "previous total supply not saved properly";
 }
 
 // passes
-rule onBurn() {
+rule burn_decreases_totalSupply() {
     env e;
-    uint256 amount;
-    address account;
+    uint256 amount; address account;
 
     uint256 fromBlock = e.block.number;
-    uint224 totalVotesBefore = totalVotes();
-    uint256 totalSupplyBefore = totalSupply();
+    uint256 totalSupply_ = totalSupply();
+
+    burn(e, account, amount);
 
-    _burn(e, account, amount);
+    uint256 _totalSupply = totalSupply();
 
-    assert totalVotes() == totalVotesBefore, "totalVotes decreased";
-    assert getPastTotalSupply(e, fromBlock) == totalSupplyBefore , "previous total supply not saved properly";
+    assert _totalSupply == totalSupply_ - amount, "totalSupply not decreased properly";
+    assert getPastTotalSupply(e, fromBlock) == totalSupply_ , "previous total supply not saved properly";
 }
+
+
+
+// passes
+rule mint_doesnt_increase_totalVotes() {
+    env e;
+    uint256 amount; address account;
+
+    uint224 totalVotes_ = totalVotes();
+
+    mint(e, account, amount);
+
+    assert totalVotes() == totalVotes_, "totalVotes increased";
+}
+// passes
+rule burn_doesnt_decrease_totalVotes() {
+    env e;
+    uint256 amount; address account;
+
+    uint224 totalVotes_ = totalVotes();
+
+    burn(e, account, amount);
+
+    assert totalVotes() == totalVotes_, "totalVotes decreased";
+}
```

Comment:
I think this part of the rewrite uses an old version.

I believe I wrote these rules, and the PR is reverting back some of the latest modifications.

In particular, the total supply check is already part of the ERC20 specs, only the total vote is relevant here.

I'm also puzzeled about what these rules say. The total number of vote should change when minting and burning to/from an account that delegated